src/reference/workflow_types.md

```diff
@@ -0,0 +1,60 @@
+# Workflow Data Types
+
+::: info
+[Learn about the concepts behind this.](/concepts/workflows/flow.md)
+:::
+
+``` rust
+pub enum NodeDataProp {
```

Comment:
Same comment as for concept, we dont want to just dump a rust enum here. We want a real reference.
Also what is more important here is the automatic conversion between types which is not present.